python/hail/tests/test_typecheck2.py

```diff
@@ -0,0 +1,367 @@
+from hail.typecheck2 import typecheck
+import unittest
+from typing import *
+
+
+class Tests(unittest.TestCase):
+
+    def test_noargs(self):
+        def f_noargs():
```

Comment:
Sorry, missed this before. Is this supposed to have a `typecheck(f_noargs)`?

Reply:
yes!